dowhy/causal_model.py

```diff
@@ -101,6 +101,8 @@ def __init__(self, data, treatment, outcome, graph=None,
                 self._graph = None
 
         else:
+            if isinstance(graph, str):
```

Comment:
why is this needed again? `init_graph` also calls CausalGraph constructor, so it should work, right?